raft/src/main/java/org/apache/kafka/raft/ResignedState.java

```diff
@@ -140,10 +140,21 @@ public List<ReplicaKey> preferredSuccessors() {
     }
 
     @Override
-    public boolean canGrantVote(ReplicaKey candidateKey, boolean isLogUpToDate) {
+    public boolean canGrantVote(ReplicaKey replicaKey, boolean isLogUpToDate) {
         log.debug(
-            "Rejecting vote request from candidate ({}) since we have resigned as candidate/leader in epoch {}",
-            candidateKey,
+            "Rejecting Vote request from candidate ({}) since we have resigned as leader in epoch {}",
+            replicaKey,
+            epoch
+        );
+
+        return false;
+    }
+
+    @Override
+    public boolean canGrantPreVote(ReplicaKey replicaKey, boolean isLogUpToDate) {
```

Comment:
Did we agree that canGrantPreVote is true in the resigned state if the log is up to date?

Reply:
we did, I updated the KIP description but not the code for this case, thanks for the catch